Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -145,31 +146,37 @@ public void Install(IWindsorContainer container, IConfigurationStore store)
 
             RegisterSpecialFactories(container);
             RegisterFactories(container, assembliesToRegister);
-            
+
             ApplyDefaultInterfaceConvention(container, assembliesToRegister);
         }
 
         // note: settings namespace classes are injected in singleton scope
         private void RegisterConfiguration(IWindsorContainer container, Assembly[] assembliesToRegister)
         {
+            var experimentalTypes = new List<Type>();
             foreach (var assembly in assembliesToRegister)
             {
-                var assemblyTypes = GetIoCRegisteredTypes(assembly);
-
-                container.Register(Classes.From(assemblyTypes)
-                    .InSameNamespaceAs<Configuration>()
+                container.Register(Classes.FromAssembly(assembly)
+                    .IncludeNonPublicTypes()
+                    .Where(type => type.NotDisabledExperimental(_initialSettings) && type.Namespace == typeof(Configuration).Namespace)
                     .WithService.AllInterfaces()
                     .LifestyleSingleton());
+
+                experimentalTypes.AddRange(assembly.GetTypes()
+                    .Where(t => Attribute.IsDefined(t, typeof(ExperimentalAttribute))));
             }
 
+            container.Register(Component.For(typeof(IEnumerable<Type>))
+                .DependsOn(Dependency.OnComponent<ViewModelBase, GeneralSettingsViewModel>())
```

Comment:
That looks the wrong way around. If you want to inject only into these types, you have to register these types separately paasing in the `IEnumerable<Type>` instance via `DependsOn`.

Reply:
For better or worse, the ViewModels are not currently injected via DI...